dll/win32/browseui/shellfind/CFindFolder.cpp

```diff
@@ -341,8 +350,30 @@ LRESULT CFindFolder::StartSearch(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL &
     pSearchData->szFileName = pSearchParams->szFileName;
     pSearchData->szQueryA = pSearchParams->szQuery;
     pSearchData->szQueryW = pSearchParams->szQuery;
+    pSearchData->SearchHidden = pSearchParams->SearchHidden;
     SHFree(pSearchParams);
 
+    TRACE("pSearchParams->SearchHidden is '%d'.\n", pSearchData->SearchHidden);
+
+    if(pSearchData->SearchHidden)
+        SearchHiddenvalue = 1;
+    else
+        SearchHiddenvalue = 0;
+
+    regOpenResult = RegOpenKeyEx(HKEY_CURRENT_USER, L"Software\\Microsoft\\Windows\\CurrentVersion\\Explorer", 0, KEY_ALL_ACCESS, &hkey);
+    if(regOpenResult == ERROR_SUCCESS)
+    {
+        if (RegSetValueEx(hkey, L"SearchHidden", NULL, dwType, (const BYTE*)&SearchHiddenvalue, size) == ERROR_SUCCESS)
+        {
+            TRACE("SearchHidden is '%d'.\n", SearchHiddenvalue);
+            RegCloseKey(hkey);
```

Comment:
This line has to be moved outside of this sub-block.

Reply:
OK.